### DISCUSSION TITLE
Setting up Hetzner bare metal cluster

### DISCUSSION THREAD

**Discussion by Stringls** [syself/cluster-api-provider-hetzner#684] [Q&A] [answered]
Hi! I am trying to create a cluster with private network on Hetzner where CP is a Hcloud VM and 1 MD is a Hetzner BM server.

When I apply `cluster.yaml` configuration CP is successfully provisioned and it starts to provision BM. I check the logs of `HetznerBareMetalHost` and it show me this error
```
bm-0   registering   **********   ***********                                       test-cluster-md-1-   ssh reboot too slow   ssh timeout error - server has not restarted yet   17m
```
But BM is still being successfully provisioned, I guess it has influence only on speed of the provisioning. Am I able to resolve this issue?

Another point, as I mentioned I am trying to create a cluster with private network. I've seen the issue where you said that it's not supported yet, but I want to clarify the issue below is definitely related to using private network.
I install Hetzner CCM on my cluster when it's provisioned to sync the nodes and get this error in the logs of the container. The pod is being installed on MD. In `/etc/hosts` of BM server there is a line `***.***.***.*** bm-test-cluster-md-1-`. I am also able to curl the below URL.
```
Error from server: Get "https://bm-test-cluster-md-1-:10250/containerLogs/kube-system/ccm-ccm-hetzner-668497c647-rbdqf/ccm-hetzner": dial tcp: lookup bm-test-cluster-md-1- on 1.1.1.1:53: no such host
```
P.S. I've successfully created a cluster when I do not use a private network feature

So, my questions are:
1. Is the issue with SSH resolvable? Does it have influence on speed of the provisioning?
2. Is the error with DNS name resolving related to using private network and it it possible to fix it?

Thanks!

**Comment by janiskemper** [comment, marked as answer]
To 1.: This is an "internal" error. We track the progress of provisioning there. As we use ssh to provision, we have to do some trial and error in between to test whether the server is back online, has a new ssh port (after provisioning e.g.), etc. 
To conclude: This error is absolutely fine and doesn't affect you at all.
To 2: @batistein has maybe more infos.

**Comment by Stringls** [comment]
@janiskemper Thanks for the answer. I've got a response in [#cluster-api](https://kubernetes.slack.com/archives/C8TSNPY4T/p1679580523521339?thread_ts=1679577975.085089&cid=C8TSNPY4T) Kubernetes Slack channel from @batistein about my issue so closing this issue
